### PR TITLE
Fix webpack setup

### PR DESCRIPTION
**Pull request by AmirSavand**

### CODE REVIEW

File: src/app/basement/pagination/pagination_service.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 "use strict";
+import paginationTemplate from "./pagination_loadmore.html";
```
Comment (Alir3z4) [inline src/app/basement/pagination/pagination_service.js]: I just love how you can import pages and everything <3

File: src/app/basement/pagination/pagination_service.js [hunk 2]
```diff
@@ -50,3 +51,4 @@ function Pagination($resource, $rootScope) {
 }
 
 app.factory("Pagination", Pagination);
```
Comment (Alir3z4) [inline src/app/basement/pagination/pagination_service.js]: I don't think you need to register it as a factory anymore, right ?

**Comment by Alir3z4** [comment]
Just left some comments.
Also good job on making a new branch from webpackios!

File: package.json [hunk 3]
```diff
@@ -73,6 +73,8 @@
   "scripts": {
     "build": "rimraf dist && webpack --bail --progress --profile",
     "server": "webpack-dev-server --history-api-fallback --inline --progress",
+    "server-dev": "webpack-dev-server --history-api-fallback --inline --progress",
+    "server-prod": "webpack-dev-server --history-api-fallback --inline --progress",
```
Comment (AmirSavand) [inline package.json]: :heart:

File: src/app/basement/pagination/pagination_service.js (hunk 2 above)
Comment (AmirSavand) [reply src/app/basement/pagination/pagination_service.js]: @ArsalanSavand handle this.

**Comment by Alir3z4** [comment]
Upgraded to latest version of webpack https://github.com/SavandBros/gonevis-dash/pull/874/commits/f234268ed02e1638a73fbec8e7438b5e0b57dd1b

It has much more goodies and nice features.

@ArsalanSavand you can go ahead and see if you can fix that issue with medium editor

File: karma.conf.js [hunk 4]
```diff
@@ -32,15 +32,6 @@ module.exports = function (config) {
...
     },
-    plugins: [
-      'karma-phantomjs-launcher',
-      'karma-jasmine',
-      'karma-coverage',
-      'karma-sourcemap-loader',
-      'karma-webpack',
-      'karma-spec-reporter',
-      'karma-babel-preprocessor'
-    ],
```
Comment (AmirSavand) [inline karma.conf.js]: What? Why?

**Comment by AmirSavand** [comment]
:+1: @Alir3z4 

Why didn't you use the latest webpack in the first place?

**Comment by AmirSavand** [comment]
Also build failed :fox_face:

**Comment by Alir3z4** [comment]
> Why didn't you use the latest webpack in the first place?

Because the I used this template https://github.com/preboot/angularjs-webpack/blob/master/webpack.config.js just to get it working

It was using version 2.

However, upgrading to version 4+ made me to change almost all of it.
Nothing was compatible and all the plugins had to be reconfigured.


> Also build failed 🦊

Now, it's passing

File: karma.conf.js (hunk 4 above)
Comment (Alir3z4) [reply karma.conf.js]: They get picked up automatically by karma itself.

**Comment by Alir3z4** [comment]
I am done with this, let's get this merged into `webpackios!` branch and bake it into `master`, after that we have all the `on-hold` and `blocked` pull requests that need to be updated.

File: package.json (hunk 3 above)
Comment (Alir3z4) [reply package.json]: By default, it's on staging
`server-dev` and `server-prod` are just extra sugars <3

**Comment by AmirSavand** [comment]
@Alir3z4 there's nothing in https://gonevis-dash-draft-pr-874.herokuapp.com 😕

**Comment by AmirSavand** [comment]
@ArsalanSavand too many 404 errors, click on links in sidebar and see the errors in console.

The pull is incomplete and not properly setup.

**Comment by Alir3z4** [comment]
> @Alir3z4 there's nothing in https://gonevis-dash-draft-pr-874.herokuapp.com 😕

It's working now, I fixed the build step

**Comment by Alir3z4** [comment]
@AmirSavand @ArsalanSavand any ETA on this ?

**Comment by Alir3z4** [comment]
@ArsalanSavand you broken the build <3

**Comment by Alir3z4** [comment]
OK, the modals don't raise any error again, but still don't work as expected and to me it seems to be a CSS issue.
When I fire up a modal, the data/html gets added to the page (chrome inspect) and I only see it when I remove the `modal fade *` classes from the modal html.

File: src/app/tag/tag_modal/tag_modal_view.html [hunk 5]
```diff
@@ -12,7 +12,7 @@
       </div>
       <div class="modal-body">
         <form ng-submit="tag.update(form)">
-          <ng-include src="'tag/includes/tag_form.html'"></ng-include>
+          <tag-form></tag-form>
```
Comment (Alir3z4) [inline src/app/tag/tag_modal/tag_modal_view.html]: @AmirSavand This doesn't work.
It doesn't actually load anything and the modal is empty.

Comment (Alir3z4) [reply src/app/tag/tag_modal/tag_modal_view.html]: I think the data should be passed into it

Comment (ArsalanSavand) [reply src/app/tag/tag_modal/tag_modal_view.html]: This was temporary until we fix the modal issue

**Comment by Alir3z4** [comment]
Actually I made this right now.
I replaced the include with a component today.

File: webpack.config.js [hunk 6]
```diff
@@ -162,7 +162,9 @@ module.exports = {
       }
     }),
     new webpack.ProvidePlugin({
-      "window.jQuery": "jquery"
```
Comment (Alir3z4) [inline webpack.config.js]: Good work on this

File: src/app/vendor.js [hunk 7]
```diff
@@ -7,6 +7,8 @@ require("angular-cookies");
 require("angular-sanitize");
 require('@uirouter/angularjs');
 require("oclazyload");
+require('bootstrap');
```
Comment (Alir3z4) [inline src/app/vendor.js]: Oh my god!

**Comment by ArsalanSavand** [comment]
We should fix unit tests and Start View page.

**Comment by AmirSavand** [comment]
@Alir3z4 how should we get the image by path in tests? Please fix this:

```
Account getMedia should return default image FAILED
        Expected '/tmp/_karma_webpack_/7a2623e0edf319d1b47be49aecdbd02b.png' to be 'style/img/avatar.png'.
```

**Comment by AmirSavand** [comment]
@ArsalanSavand what's wrong with start page?

**Comment by Alir3z4** [comment]
The avatar spec has been fixed in eb67510

**Pull request merged**